header_compilation_tests.sh

```diff
@@ -44,21 +44,21 @@ if ! [ -z ${ROCM_PATH+x} ]; then
     rocm_path=${ROCM_PATH}
```

Comment:
This file is [not used in hip-clang builds](https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/0155bcd67cd4587f78ca1774509f387de50cc7e9/install.sh#L556), and should not be modified until it is completely ported to hip-clang, which is non-trivial. The existing commands have contextual information which must be translated into hip-clang equivalents. HCC should not be removed, but should modified to hip-clang. I reject the changes in this file.